src/MongoDB/Query.c

```diff
@@ -90,6 +90,15 @@ static bool php_phongo_query_opts_append_document(bson_t* opts, const char* opts
 		}                                                                                                            \
 	}
 
+#define PHONGO_QUERY_OPT_BOOL_DEPRECATED(opt, zarr, key)                                                                    \
```

Comment:
Rather than duplicate these macros in their entirety, what do you think of:

```c
#define PHONGO_QUERY_OPT_BOOL_EX(opt, zarr, key, deprecated)
	if ((zarr) && php_array_existsc((zarr), (key))) {
		if (deprecated) {
			php_error_docref(NULL TSRMLS_CC, E_DEPRECATED, "The \"%s\" option is deprecated and will be removed in a future release", key);
		}
		if (!BSON_APPEND_BOOL(intern->opts, (opt), php_array_fetchc_bool((zarr), (key)))) {
			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Error appending \"%s\" option", (opt));
			return false;
		}
	}

#define PHONGO_QUERY_OPT_BOOL(opt, zarr, key) PHONGO_QUERY_OPT_BOOL_EX((opt), (zarr), (key), 0)
#define PHONGO_QUERY_OPT_BOOL_DEPRECATED(opt, zarr, key) PHONGO_QUERY_OPT_BOOL_EX((opt), (zarr), (key), 1)
```

This would mitigate the risk of the macros getting out of sync.

Reply:
Done